client/js/_author/components/common/tiny_wrapper.jsx

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import TinyMCE     from 'react-tinymce';
+import guid from '../../../utils/guid';
+
+import 'tinymce';
+import 'tinymce/themes/modern/theme'
+import 'tinymce/plugins/autolink/plugin';
+import 'tinymce/plugins/paste/plugin';
+import 'tinymce/plugins/link/plugin';
+import 'tinymce/plugins/image/plugin';
+import 'tinymce/plugins/lists/plugin';
+
+import * as AssetActions from '../../../actions/qbank/assets';
+
+function select(state) {
+  return {
+    uploadedAssets: state.uploadedAssets,
+    apiUrl: state.settings.api_url,
+  };
+}
+
+export class TinyWrapper extends React.Component {
+  static propTypes = {
+    text: React.PropTypes.string,
+    uploadImage: React.PropTypes.func.isRequired,
+    bankId: React.PropTypes.string.isRequired,
+  };
+
+  constructor() {
+    super();
+
+    this.state = {
+      imageCallback: null,
+      imageGuid: null,
+      uploadingImage: false
+    };
+
+    this.id = guid();
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const imageUrl = nextProps.uploadedAssets[this.state.imageGuid];
+    if(this.state.uploadingImage && imageUrl) {
+      this.state.imageCallback(imageUrl);
+      this.setState({uploadingImage: false});
+    }
+  }
+
+  shouldComponentUpdate() {
+    // TinyMCE doesn't handle rerendering very well, and everything dealing
+    // with the content inside of it is completely managed by tinyMCE, so we
+    // never rerender. All state that needs to be updated is done so in
+    // componentWillReceiveProps or in callbacks from tinyMCE.
+    return false;
+  }
+
+  uploadImage(e) {
+    const imageGuid = guid();
```

Comment:
Turns out this is unnecessary, I thought I'd seen this callback get called without a file but I think that's impossible